src/trio/_core/_tests/test_run.py

```diff
@@ -2216,19 +2220,41 @@ def test_Nursery_subclass() -> None:
 
 
 def test_Cancelled_init() -> None:
-    with pytest.raises(TypeError):
-        raise _core.Cancelled
+    with pytest.raises(TypeError, match=r"^trio.Cancelled has no public constructor$"):
+        raise _core.Cancelled  # type: ignore[call-arg]
 
-    with pytest.raises(TypeError):
-        _core.Cancelled()
+    with pytest.raises(TypeError, match=r"^trio.Cancelled has no public constructor$"):
+        _core.Cancelled(source="explicit")
 
     # private constructor should not raise
-    _core.Cancelled._create()
-
+    _core.Cancelled._create(source="explicit")
+
+
+async def test_Cancelled_str() -> None:
+    cancelled = _core.Cancelled._create(source="explicit")
+    assert str(cancelled) == "cancelled due to explicit"
+    assert re.fullmatch(
+        r"cancelled due to deadline from task "
+        r"<Task 'trio._core._tests.test_run.test_Cancelled_str' at 0x\w*>",
```

Comment:
```suggestion
        rf"{trio.lowlevel.current_task()!r}",
```
? Same elsewhere.

Reply:
I intentionally left a few to show what the full message would look like to the user, though I suppose I could do that with a comment or a more explicit demonstration.

Reply:
added a comment to one and replaced the other